homeassistant/components/sensor/pvoutput.py

```diff
@@ -0,0 +1,121 @@
+"""
+Support for getting collected information from PVOutput.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.pvoutput/
+"""
+import logging
+from collections import namedtuple
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.components.sensor.rest import RestData
+from homeassistant.const import (
+    ATTR_TEMPERATURE, CONF_API_KEY, CONF_NAME, STATE_UNKNOWN)
+
+_LOGGER = logging.getLogger(__name__)
+_ENDPOINT = 'http://pvoutput.org/service/r2/getstatus.jsp'
+
+ATTR_DATE = 'date'
+ATTR_TIME = 'time'
+ATTR_ENERGY_GENERATION = 'energy_generation'
+ATTR_POWER_GENERATION = 'power_generation'
+ATTR_ENERGY_CONSUMPTION = 'energy_consumption'
+ATTR_POWER_CONSUMPTION = 'power_consumption'
+ATTR_EFFICIENCY = 'efficiency'
+ATTR_VOLTAGE = 'voltage'
+
+CONF_SYSTEM_ID = 'system_id'
+
+DEFAULT_METHOD = 'GET'
+DEFAULT_NAME = 'PVOutput'
+DEFAULT_VERIFY_SSL = True
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_API_KEY): cv.string,
+    vol.Required(CONF_SYSTEM_ID): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the PVOutput sensor."""
+    name = config.get(CONF_NAME)
+    api_key = config.get(CONF_API_KEY)
+    system_id = config.get(CONF_SYSTEM_ID)
+    method = 'GET'
+    payload = None
+    auth = None
+    headers = {
+        'X-Pvoutput-Apikey': api_key,
+        'X-Pvoutput-SystemId': system_id,
+    }
+    verify_ssl = DEFAULT_VERIFY_SSL
```

Comment:
This flag is only needed when you want to connect over HTTPS but do not want to validate the certificate (basically rendering HTTPS useless)

If they only use HTTP, just use the HTTP protocol and forget about this flag.